Quarkus 3: Model tests fail to finish

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

testsuite

### Describe the bug

Model tests in `quarkus3` branch do not finish in time, see e.g. https://github.com/keycloak/keycloak/actions/runs/3958440904/jobs/6781996089

They repeatedly fail with

```
java.lang.NoClassDefFoundError: javax/transaction/RollbackException
   at org.infinispan.client.hotrod.RemoteCacheManager.<init>(RemoteCacheManager.java:171)
   at org.infinispan.client.hotrod.RemoteCacheManager.<init>(RemoteCacheManager.java:131)
   at org.keycloak.models.map.storage.hotRod.connections.DefaultHotRodConnectionProviderFactory.lazyInit(DefaultHotRodConnectionProviderFactory.java:127)
   at org.keycloak.models.map.storage.hotRod.connections.DefaultHotRodConnectionProviderFactory.create(DefaultHotRodConnectionProviderFactory.java:70)
   at org.keycloak.models.map.storage.hotRod.connections.DefaultHotRodConnectionProviderFactory.create(DefaultHotRodConnectionProviderFactory.java:54)
   at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:271)
   at org.keycloak.models.map.storage.hotRod.HotRodMapStorageProviderFactory.createHotRodStorage(HotRodMapStorageProviderFactory.java:191)
   at org.keycloak.models.map.storage.hotRod.HotRodMapStorageProviderFactory.lambda$getHotRodStorage$0(HotRodMapStorageProviderFactory.java:187)
   at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
   at org.keycloak.models.map.storage.hotRod.HotRodMapStorageProviderFactory.getHotRodStorage(HotRodMapStorageProviderFactory.java:187)
   at org.keycloak.models.map.storage.hotRod.HotRodMapStorageProvider.getStorage(HotRodMapStorageProvider.java:38)
   at org.keycloak.models.map.common.AbstractMapProviderFactory.getStorage(AbstractMapProviderFactory.java:112)
   at org.keycloak.models.map.realm.MapRealmProviderFactory.createNew(MapRealmProviderFactory.java:35)
   at org.keycloak.models.map.realm.MapRealmProviderFactory.createNew(MapRealmProviderFactory.java:27)
   at org.keycloak.models.map.common.AbstractMapProviderFactory.create(AbstractMapProviderFactory.java:98)
   at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:271)
   at org.keycloak.models.map.datastore.MapDatastoreProvider.realms(MapDatastoreProvider.java:59)
   at org.keycloak.services.DefaultKeycloakSession.realms(DefaultKeycloakSession.java:367)
   at org.keycloak.testsuite.model.KeycloakModelTest.createRealm(KeycloakModelTest.java:631)
   at org.keycloak.testsuite.model.events.EventQueryTest.createEnvironment(EventQueryTest.java:52)
   at org.keycloak.models.utils.KeycloakModelUtils.lambda$runJobInTransaction$1(KeycloakModelUtils.java:256)
   at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:269)
   at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:255)
   at org.keycloak.testsuite.model.KeycloakModelTest.createEnvironment(KeycloakModelTest.java:532)
   at jdk.internal.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
   at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
   at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
   at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
   at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
   at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
   at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
   at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
   at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
   at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
   at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
   at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
   at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
   at org.junit.rules.RunRules.evaluate(RunRules.java:20)
   at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
   at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
   at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
   at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:302)
   at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
   at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
   at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
   at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
   at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
   at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: java.lang.ClassNotFoundException: javax.transaction.RollbackException
   at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
   at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
   at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
   ... 56 more
```

### Version

quarkus3

### Expected behavior

Model tests finish

### Actual behavior

Timeout

### How to Reproduce?

Run model tests in GHA

### Anything else?

_No response_

--- a/model/infinispan/pom.xml
+++ b/model/infinispan/pom.xml
@@ -57,7 +57,17 @@
         </dependency>
         <dependency>
             <groupId>org.infinispan</groupId>
+            <artifactId>infinispan-client-hotrod-jakarta</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.infinispan</groupId>
             <artifactId>infinispan-cachestore-remote</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.infinispan</groupId>
+                    <artifactId>infinispan-client-hotrod</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.infinispan</groupId>

--- a/testsuite/model/pom.xml
+++ b/testsuite/model/pom.xml
@@ -108,7 +108,7 @@
         </dependency>
         <dependency>
             <groupId>org.infinispan</groupId>
-            <artifactId>infinispan-server-core</artifactId>
+            <artifactId>infinispan-core-jakarta</artifactId>
         </dependency>
         <dependency>
             <groupId>org.postgresql</groupId>

